app/gui2/src/components/ComponentBrowser.vue

```diff
@@ -19,52 +37,301 @@ function positionAtMouse(): boolean {
   return true
 }
 
-const transform = computed(() => {
-  const nav = props.navigator
-  const pos = scenePosition.value
-  return `${nav.transform} translate(${pos.x}px, ${pos.y}px) scale(${
-    1 / nav.scale
-  }) translateY(-100%)`
+// === Components List and Positions ===
+
+const componentStore = useComponentsStore()
+
+const visibleComponents = computed(() => {
+  if (scroller.value) {
+    const first_visible = componentAtY(animatedScrollPosition.value)
+    const last_visible = componentAtY(animatedScrollPosition.value + scrollerSize.value.y)
+    return componentStore.components.slice(first_visible, last_visible + 1).map((component, i) => {
+      return { component, index: i + first_visible }
+    })
+  } else {
+    return []
+  }
 })
 
-useWindowEvent('keypress', (e) => {
+function componentPos(index: number) {
+  return index * ITEM_SIZE
+}
+
+function componentAtY(pos: number) {
+  return Math.floor(pos / ITEM_SIZE)
+}
+
+function componentStyle(index: number) {
+  return { transform: `translateY(${componentPos(index)}px)` }
+}
+
+function componentColor(component: Component): string {
+  return componentStore.groups[component.group].color
+}
+
+// === Highlight ===
+
+const selected = ref(-1)
+const selectedPosition = computed(() => componentPos(selected.value))
+const highlightPosition = useApproach(selectedPosition)
+
+const highlightClipPath = computed(() => {
+  let top = highlightPosition.value
+  let bottom = listContentHeight.value - top - ITEM_SIZE
+  return `inset(${top}px 0px ${bottom}px 0px round 16px)`
+})
+
+function navigateFirst() {
+  selected.value = 0
+  scrollToSelected()
+}
+
+function navigateLast() {
+  selected.value = componentStore.components.length - 1
+  scrollToSelected()
+}
+
+function navigateUp() {
+  if (selected.value > 0) {
+    selected.value -= 1
+  }
+  scrollToSelected()
+}
+
+function navigateDown() {
+  if (selected.value < componentStore.components.length - 1) {
+    selected.value += 1
+  }
+  scrollToSelected()
+}
+
+// === Scrolling ===
+
+const scroller = ref<HTMLElement>()
+const scrollerSize = useResizeObserver(scroller)
+const scrollPosition = ref(123)
+const animatedScrollPosition = useApproach(scrollPosition)
+
+const listContentHeight = computed(() => componentStore.components.length * ITEM_SIZE)
+const listContentHeightPx = computed(() => `${listContentHeight.value}px`)
+
+function scrollToSelected() {
+  scrollPosition.value = selectedPosition.value - scrollerSize.value.y + ITEM_SIZE
+}
+
+function updateScroll() {
+  if (scroller.value && Math.abs(scroller.value.scrollTop - animatedScrollPosition.value) > 1.0) {
+    scrollPosition.value = scroller.value.scrollTop
+    animatedScrollPosition.skip()
+  }
+}
+
+// === Documentation Panel ===
+
+const docs_visible = ref(true)
```

Comment:
Inconsistent variable casing. This is (unfortunately) not rust.

Reply:
Can we enable some lint checking it?

Reply:
Rule has been added in my PR and is now merged to develop.